pallets/parachain-system/src/lib.rs

```diff
@@ -235,10 +238,24 @@ decl_module! {
 			let _ = Self::send_upward_message(message);
 		}
 
-		#[weight = (1_000, DispatchClass::Operational)]
-		fn sudo_send_hrmp_message(origin, message: OutboundHrmpMessage) {
+		#[weight = (1_000_000, DispatchClass::Operational)]
+		fn authorize_upgrade(origin, code_hash: T::Hash) {
 			ensure_root(origin)?;
-			let _ = Self::send_hrmp_message(message);
+
+			AuthorizedUpgrade::<T>::put(&code_hash);
+
+			Self::deposit_event(RawEvent::UpgradeAuthorized(code_hash));
+		}
+
+		#[weight = 1_000_000]
+		fn enact_authorized_upgrade(origin, code: Vec<u8>) {
```

Comment:
```suggestion
		fn enact_authorized_upgrade(_origin, code: Vec<u8>) {
```

Should we maybe pay back the fees if the function was successful?

Reply:
```
-> DispatchResultWithPostInfo
```

and

```
Ok(Pays::No.into())
```